dialog_design.md

```diff
@@ -0,0 +1,68 @@
+# Dialogs
+
+Goal: Display a message and an optional graphic to all users, or some, or one, and return a response.
+
+## Architecture
+
+Create a new `Dialog` class that manages calls and responses. It is also a good place to park static convenience
```

Comment:
>Create a new `Dialog` class that manages calls and responses [](start = 0, length = 60)

I think it should an actor, like text (in the future a component).
 a. solves ownership, destruction, transform.
 b. solves user access - single-user/groups/everyone
 c. we could add an option to close on selection (default to on) - leave the dialog up after clicks, or just auto-close.
 d. we could add an option to locally user-face it (that's one reason why it's a child of an actor, not an actor - it's per-user only)
 e. solves your "persistent" suggestion - it's automatically synced to new users that meet the filtering requirements.
  f. could have option to just re-show the dialog.

Reply:
It seems like a stretch to call something with this functionality an Actor. There's no transform (it's modal), there's no hierarchy, and there's no global state to synchronize. None of the other Actor components would make sense on a dialog (collider, rigidbody, mesh, light...). If anything, this is the first piece of our UI infrastructure, but even that is a stretch since it's unsynchronized by necessity (how weird would it be if you could see other people type into your modal dialog?)

---
In reply to: [321512104](https://github.com/microsoft/mixed-reality-extension-sdk/pull/444#discussion_r321512104) [](ancestors = 321512104)

Reply:
I think it would be wrong to make this an Actor component. This is local UI. It should never be synced or visible to other players.
- The dialog shouldn't have a transform. It should be view-locked.
- single-user/groups/everyone is already solved in this design.
- when new users join, the app should decide if they should see the dialog and can show it to them. This would be straight forward to do in the userJoined callback.

Soren did you have a use case in mind where an Actor-based dialog was a better solution?